src/components/toolbar/saved-notebooks-and-examples-subsection.jsx

```diff
@@ -1,47 +1,63 @@
+import { connect } from 'react-redux'
+
 import NotebookMenuItem from './notebook-menu-item'
 import NotebookMenuDivider from './notebook-menu-divider'
 import NotebookMenuHeader from './notebook-menu-header'
 import NotebookMenuSubsection from './notebook-menu-subsection'
 
+
 import iodideExampleTasks from '../../iodide-examples'
 import tasks, { getLocalStorageNotebook } from '../../task-definitions'
 import { stateFromJsmd } from '../../jsmd-tools'
 
-const AUTOSAVE = 'AUTOSAVE: '
-const autosave = Object.keys(localStorage).filter(n => n.includes(AUTOSAVE))[0] //
-const locallySaved = Object.keys(localStorage).filter(n => !n.includes(AUTOSAVE))
+class SavedNotebooksAndExamplesSubsection extends React.Component {
+  static propTypes = {
+    locallySaved: PropTypes.array,
+    autoSave: PropTypes.string,
+  }
 
-let autoSaveMenuItem
-let autoSave
-if (autosave !== undefined) {
-  autoSave = getLocalStorageNotebook(autosave)
-  autoSaveMenuItem = <NotebookMenuItem task={autoSave} key={autoSave.title} />
-}
+  getLocallySavedMenuItems() {
+    let locallySavedMenuItems
+    if (this.props.locallySaved) {
+      this.props.locallySaved.sort((a, b) => {
+        const p = (_) => {
+          let ls = localStorage.getItem(_)
+          if (!ls) return -1
+          ls = stateFromJsmd(ls)
+          return Date.parse(ls.lastSaved)
+        }
+        return p(b) - p(a)
+      })
 
-locallySaved.sort((a, b) => {
-  const p = (_) => {
-    let ls = localStorage.getItem(_)
-    if (!ls) return -1
-    ls = stateFromJsmd(ls)
-    return Date.parse(ls.lastSaved)
+      if (this.props.locallySaved.length) {
+        locallySavedMenuItems = this.props.locallySaved.map((l) => {
+          const task = getLocalStorageNotebook(l)
+          if (task !== undefined) {
+            return <NotebookMenuItem task={task} key={task.title} />
+          }
+          return undefined
+        })
+      }
+    }
+    return locallySavedMenuItems
   }
-  return p(b) - p(a)
-})
-
-let locallySavedMenuItems
-if (locallySaved.length) {
-  locallySavedMenuItems = locallySaved.map((l) => {
-    const task = getLocalStorageNotebook(l)
-    if (task !== undefined) {
-      return <NotebookMenuItem task={task} key={task.title} />
+
+  getAutoSaveMenuItem() {
+    let autoSave
+    let autoSaveMenuItem
+    if (this.props.autoSave) {
+      autoSave = getLocalStorageNotebook(this.props.autoSave)
+      autoSaveMenuItem = <NotebookMenuItem task={autoSave} key={autoSave.title} />
     }
-    return undefined
-  })
-}
+    return autoSaveMenuItem
+  }
 
-export default class SavedNotebooksAndExamplesSubsection extends React.Component {
   render() {
+    const autoSaveMenuItem = this.getAutoSaveMenuItem()
+    const locallySavedMenuItems = this.getLocallySavedMenuItems()
+
     return (
       <NotebookMenuSubsection title="notebooks ... " {...this.props} >
         {autoSaveMenuItem ? <NotebookMenuHeader key="autosave" title="Auto-Saved" /> : undefined}
```

Comment:
as long as you're in here, you maybe you could replace this 
```
( x ? a )
( x || b )
( x ? c )
```
pattern with more explicit if/else blocks

:-)